### ISSUE TITLE
Kube-Controller-Manager failed to start after restart the whole Kuberetes cluster when external APIService (such as service catalog) is deployed and registred

### ISSUE THREAD

**Issue by luckyfengyong** [kubernetes/kubernetes#53424]
**Is this a BUG REPORT or FEATURE REQUEST?**:

/kind bug

**What happened**:
After deploying service catalog (or other external APIService) into Kuberenetes, and then restart the whole kubernetes cluster (or Kuberetnes master if service catalog api is running in the master node), Kube-Controller-Manager failed to start.

**What you expected to happen**:
The Kube-Controller-Manager is able to start and continue working

**How to reproduce it (as minimally and precisely as possible)**:
1. Create a one node kubernetes cluster with calico
2. Follow the instruction of [service catalog helm charts](https://github.com/kubernetes-incubator/service-catalog/tree/master/charts/catalog) to deploy service catalog
3. Restart docker service

**Anything else we need to know?**:
Kube-Controller-Manager failed to start with following error message 
```
F0921 06:47:47.847921       1 controllermanager.go:176] error building controller context: failed to get supported resources from server: unable to retrieve the complete list of server APIs: servicecatalog.k8s.io/v1alpha1: an error on the server ("Error: 'dial tcp 10.0.0.180:443: i/o timeout'\nTrying to reach: 'https://10.0.0.180:443/apis/servicecatalog.k8s.io/v1alpha1'") has prevented the request from succeeding
```
After containers are restart, the execution flow of cluster recovery is

1. The container of catalog api get the new pod ip
2. kubelet update the pod ip to kube api service and then etcd
(2a). kube controller manager is expected to get the update of pod to update the endpoint
(2b). kubeproxy is expected to get the update of the endpiont to update the iptable
3. kube controller manager start up and failed during initialization as it cannot talk to catalog api service.

The `2a`/`2b` and `3` become circle dependence. 

**Environment**:
- Kubernetes version (use `kubectl version`): v 1.7.1
- Cloud provider or hardware configuration**: vagrant and virtual box
- OS (e.g. from /etc/os-release): 
```
NAME="Ubuntu"
VERSION="16.04.2 LTS (Xenial Xerus)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 16.04.2 LTS"
VERSION_ID="16.04"
HOME_URL="http://www.ubuntu.com/"
SUPPORT_URL="http://help.ubuntu.com/"
BUG_REPORT_URL="http://bugs.launchpad.net/ubuntu/"
VERSION_CODENAME=xenial
UBUNTU_CODENAME=xenial
```
- Kernel (e.g. `uname -a`): ```Linux cfc6 4.4.0-64-generic #85-Ubuntu SMP Mon Feb 20 11:50:30 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux```

**Comment by liggitt** [comment]
Fixed in https://github.com/kubernetes/kubernetes/pull/49495

Fixed in 1.7.5

**Comment by luckyfengyong** [comment]
Thanks @liggitt
